modules/analysis-common/src/main/java/org/opensearch/analysis/common/SynonymTokenFilterFactory.java

```diff
@@ -64,8 +66,16 @@ public class SynonymTokenFilterFactory extends AbstractTokenFilterFactory {
     protected final Settings settings;
     protected final Environment environment;
     protected final AnalysisMode analysisMode;
-
-    SynonymTokenFilterFactory(IndexSettings indexSettings, Environment env, String name, Settings settings) {
+    private final String synonymAnalyzer;
```

Comment:
Nitpick -- maybe name this `synonymAnalyzerName`, so it's clear that it's not the analyzer itself, but the `String` name that will be looked up from the registry.

Reply:
Sure let me update this.

Reply:
Addressed in my latest commit https://github.com/opensearch-project/OpenSearch/compare/db5799521b4fb9f4e3082c46ba0c556fa03b41d2..84de9dc5dea3bb0d4310617e54b3ad45444cbe0c.